File: src/main/java/com/vjiki/music/mapper/UserMapper.java
package com.vjiki.music.mapper;

import com.vjiki.music.dto.UserResponse;
import com.vjiki.music.entity.User;

public final class UserMapper {

    private UserMapper() {
    }

    public static UserResponse toResponse(User user) {
        return UserResponse.builder()
                .id(user.getId())
                .email(user.getEmail())
                .nickname(user.getNickname())
                .avatarUrl(user.getAvatarUrl())
                .accessLevel(user.getAccessLevel() == null ? null : user.getAccessLevel().name())
                .isActive(user.getIsActive())
                .isVerified(user.getIsVerified())
                .lastLoginAt(user.getLastLoginAt())
                .createdAt(user.getCreatedAt())
                .build();
    }
}

File: src/main/kotlin/com/vjiki/music/mapper/UserMapper.kt
package com.vjiki.music.mapper

import com.vjiki.music.dto.UserResponse
import com.vjiki.music.entity.User
import org.springframework.stereotype.Component

@Component
object UserMapper {
    fun User.toResponse(): UserResponse {
        return UserResponse(
            id = id,
            email = email,
            nickname = nickname,
            avatarUrl = avatarUrl,
            accessLevel = accessLevel.name,
            isActive = isActive,
            isVerified = isVerified,
            lastLoginAt = lastLoginAt,
            createdAt = createdAt
        )
    }
}

